### PR TITLE
Add turn-on and change-mode delays.

### PR DESCRIPTION
**Pull request by reitermarkus**

### CODE REVIEW

**Review by eldruin — COMMENTED** [review]
The delay change looks good to me, thank you!
Could you:
- Document why the delay is necessary.
- Document that you can use a `NoopDelay` trait implementation (with a small example) if you want to opt out of this.
- Add an entry to the changelog.

**Comment by reitermarkus** [comment]
Added some documentation and a changelog entry.

File: src/delay.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+The given `delay` is used to wait for the sensor to turn on or change modes,
+according to the times specified in Table 14 and Table 15 in the [datasheet].
+You can opt out of this by using a no-op delay implementation to, see
```
Comment (eldruin) [inline src/delay.md:3]:
```suggestion
You can opt out of this by using a no-op delay implementation too. See
```

**Review by eldruin — COMMENTED** [review]
Looks good to me! just a small typo left and a rebase after the merge of #12.

**Comment by eldruin** [comment]
Hmm CI has a couple of problems, see [here](https://github.com/eldruin/lsm303agr-rs/runs/7721561332)

**Review by eldruin — APPROVED** [review]
Great, thank you!

**Pull request merged**